Cargo.toml

```diff
@@ -9,36 +9,36 @@ repository = "https://github.com/stackabletech/operator-rs"
 
 [dependencies]
 chrono = { version = "0.4.26", default-features = false }
-clap = { version = "4.3.4", features = ["derive", "cargo", "env"] }
+clap = { version = "4.3.19", features = ["derive", "cargo", "env"] }
 const_format = "0.2.31"
-either = "1.8.1"
+either = "1.9.0"
 futures = "0.3.28"
 json-patch = "1.0.0"
-k8s-openapi = { version = "0.18.0", default-features = false, features = ["schemars", "v1_26"] }
-kube = { version = "0.83.0", features = ["jsonpatch", "runtime", "derive"] }
+k8s-openapi = { version = "0.19.0", default-features = false, features = ["schemars", "v1_27"] }
+kube = { version = "0.85.0", features = ["jsonpatch", "runtime", "derive"] }
 lazy_static = "1.4.0"
 product-config = { git = "https://github.com/stackabletech/product-config.git", tag = "0.4.0" }
 rand = "0.8.5"
-regex = "1.8.4"
+regex = "1.9.3"
 schemars = "0.8.12"
-serde = { version = "1.0.164", features = ["derive"] }
-serde_json = "1.0.96"
-serde_yaml = "0.9.21"
-strum = { version = "0.24.1", features = ["derive"] }
-thiserror = "1.0.40"
-tokio = { version = "1.28.2", features = ["macros", "rt-multi-thread"] }
+serde = { version = "1.0.183", features = ["derive"] }
+serde_json = "1.0.104"
+serde_yaml = "0.9.25"
+strum = { version = "0.25.0", features = ["derive"] }
+thiserror = "1.0.44"
+tokio = { version = "1.29.1", features = ["macros", "rt-multi-thread"] }
 tracing = "0.1.37"
 tracing-subscriber = { version = "0.3.17", features = ["env-filter"] }
 derivative = "2.2.0"
-tracing-opentelemetry = "0.19.0"
-opentelemetry = { version = "0.19.0", features = ["rt-tokio"] }
-opentelemetry-jaeger = { version = "0.18.0", features = ["rt-tokio"] }
+tracing-opentelemetry = "0.20.0"
+opentelemetry = { version = "0.20.0", features = ["rt-tokio"] }
+opentelemetry-jaeger = { version = "0.19.0", features = ["rt-tokio"] }
 stackable-operator-derive = { path = "stackable-operator-derive" }
-snafu = "0.7.4"
+snafu = "0.7.5"
```

Comment:
We currently depend on `thiserror` and `snafu`. I think we should get rid of `snafu` and use `thiserror` exclusively. In the PR https://github.com/stackabletech/stackable-cockpit/pull/55 I'm also replacing `snafu` with `thiserror` as it better integrates with other crates from the community.

Reply:
No objections, I don't know them well enough. Is this crystal clear or should we discuss this in Arch meeting?

Reply:
Have added it to the Arch meeting agenda. I would like to address this in a dedicated PR if that's ok for you

Reply:
> Is this crystal clear or should we discuss this in Arch meeting?

For me it is, but others might have some additional input on this...

> I would like to address this in a dedicated PR if that's ok for you

Yes, totally!